sdk/test/trace/tracer_test.cc

```diff
@@ -118,28 +164,41 @@ TEST(Tracer, StartSpanWithOptionsTime)
 
 TEST(Tracer, StartSpanWithAttributes)
 {
-  std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received(
+  std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received_1(
       new std::vector<std::unique_ptr<SpanData>>);
-  auto tracer = initTracer(spans_received);
+  // The default tracer has empty sampling result attribute
+  auto tracer_default = initTracer(spans_received_1);
 
   {
-    tracer->StartSpan("span 1", {{"attr1", 314159}, {"attr2", false}, {"attr1", "string"}});
+    tracer_default->StartSpan("span 1", {{"attr1", 314159}, {"attr2", false}, {"attr1", "string"}});
 
     std::map<std::string, common::AttributeValue> m;
     m["attr3"] = 3.0;
-    tracer->StartSpan("span 2", m);
+    tracer_default->StartSpan("span 2", m);
   }
 
-  ASSERT_EQ(2, spans_received->size());
+  ASSERT_EQ(2, spans_received_1->size());
 
-  auto &span_data = spans_received->at(0);
+  auto &span_data = spans_received_1->at(0);
   ASSERT_EQ(2, span_data->GetAttributes().size());
   ASSERT_EQ("string", nostd::get<nostd::string_view>(span_data->GetAttributes().at("attr1")));
   ASSERT_EQ(false, nostd::get<bool>(span_data->GetAttributes().at("attr2")));
 
-  auto &span_data2 = spans_received->at(1);
+  auto &span_data2 = spans_received_1->at(1);
   ASSERT_EQ(1, span_data2->GetAttributes().size());
   ASSERT_EQ(3.0, nostd::get<double>(span_data2->GetAttributes().at("attr3")));
+
+  std::shared_ptr<std::vector<std::unique_ptr<SpanData>>> spans_received_2(
+      new std::vector<std::unique_ptr<SpanData>>);
+  // The default tracer has two elements in the sampling result attribute
+  auto tracer_mock = initTracer(spans_received_2, std::make_shared<MockSampler>());
+  {
+    tracer_mock->StartSpan("span 3", {{"attr1", 314159}, {"attr2", false}});
+  }
+  auto &span_data3 = spans_received_2->at(0);
+  // It has a total of four attributes: two from StartSpan and two from the sampler.
+  ASSERT_EQ(4, span_data3->GetAttributes().size());
```

Comment:
Please add tests for the values here, to be consistent with the test statements above.